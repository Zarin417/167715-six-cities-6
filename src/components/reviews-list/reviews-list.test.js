import React from "react";
import {render} from "@testing-library/react";
import ReviewsList from "./reviews-list";

it(`ReviewsList should render correctly`, () => {
  const reviews = [
    {
      offerId: 1,
      comment: `Good`,
      date: `2019-09-08T14:13:56.569Z`,
      id: 1,
      rating: 4,
      user: {
        avatarUrl: `img/avatar-max.jpg`,
        id: 4,
        isPro: false,
        name: `Max`
      }
    },
    {
      offerId: 1,
      comment: `Cozy`,
      date: `2020-06-08T14:13:56.569Z`,
      id: 2,
      rating: 4,
      user: {
        avatarUrl: `img/avatar-max.jpg`,
        id: 5,
        isPro: true,
        name: `Alex`
      }
    },
    {
      offerId: 2,
      comment: `Perfect`,
      date: `2020-05-28T14:13:56.569Z`,
      id: 3,
      rating: 5,
      user: {
        avatarUrl: `img/avatar-max.jpg`,
        id: 6,
        isPro: true,
        name: `Frederik`
      }
    }
  ];

  const {container} = render(
      <ReviewsList reviews={reviews} />
  );

  expect(container).toMatchSnapshot();
});
