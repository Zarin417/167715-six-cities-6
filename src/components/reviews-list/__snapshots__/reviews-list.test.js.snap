// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewsList should render correctly 1`] = `
<div>
  <ul
    class="reviews__list"
  >
    <li
      class="reviews__item"
    >
      <div
        class="reviews__user user"
      >
        <div
          class="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            class="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          class="reviews__user-name"
        >
          Alex
        </span>
        <span
          class="property__user-status"
        >
          Pro
        </span>
      </div>
      <div
        class="reviews__info"
      >
        <div
          class="reviews__rating rating"
        >
          <div
            class="reviews__stars rating__stars"
          >
            <span
              style="width: 80%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          class="reviews__text"
        >
          Cozy
        </p>
        <time
          class="reviews__time"
          datetime="2020-06-08"
        >
          June 8
        </time>
      </div>
    </li>
    <li
      class="reviews__item"
    >
      <div
        class="reviews__user user"
      >
        <div
          class="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            class="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          class="reviews__user-name"
        >
          Frederik
        </span>
        <span
          class="property__user-status"
        >
          Pro
        </span>
      </div>
      <div
        class="reviews__info"
      >
        <div
          class="reviews__rating rating"
        >
          <div
            class="reviews__stars rating__stars"
          >
            <span
              style="width: 100%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          class="reviews__text"
        >
          Perfect
        </p>
        <time
          class="reviews__time"
          datetime="2020-05-28"
        >
          May 28
        </time>
      </div>
    </li>
    <li
      class="reviews__item"
    >
      <div
        class="reviews__user user"
      >
        <div
          class="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            class="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          class="reviews__user-name"
        >
          Max
        </span>
      </div>
      <div
        class="reviews__info"
      >
        <div
          class="reviews__rating rating"
        >
          <div
            class="reviews__stars rating__stars"
          >
            <span
              style="width: 80%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          class="reviews__text"
        >
          Good
        </p>
        <time
          class="reviews__time"
          datetime="2019-09-08"
        >
          September 8
        </time>
      </div>
    </li>
  </ul>
</div>
`;
